src/bcccp/carpark/exit/ExitController.java

```diff
@@ -20,40 +19,86 @@
 	private IAdhocTicket  adhocTicket = null;
 	private long exitTime;
 	private String seasonTicketId = null;
-	
-	
 
-	public ExitController(Carpark carpark, IGate exitGate, 
-			ICarSensor is,
-			ICarSensor os, 
-			IExitUI ui) {
-		//TODO Implement constructor
-	}
+	private final long FIFTEEN_MINUTES = 900000; //milliseconds
+
+
+	public ExitController(Carpark carpark, IGate exitGate, ICarSensor is, ICarSensor os, IExitUI ui) {
+
+		this.carpark = carpark;
+
+		this.exitGate = exitGate;
+
+		insideSensor = is;
+
+		outsideSensor = os;
 
+		this.ui = ui;
+
+	}
 
 
+	// STEP: Read barcode.
+// The bar code is read and a check is made that no more than 15 minutes have elapsed.
```

Comment:
Please let me know when the code is fixed so Wayne can review it again. After the second review and if everything is OK, we can Merge with Master. 
Cheers,